### PR TITLE
added exception logging to ParallelIterator and logging at every 100 …

### PR DESCRIPTION
**Pull request by Kiko-Aumond**
…successful samples

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Random-Word** [comment]
Is it possible to make the log that says we've received a batch of samples also report the target count and how many have been accumulated towards that count? Thank you!

**Comment by Kiko-Aumond** [comment]
> 

Yes, let me do that.

File: rllib/execution/rollout_ops.py [hunk 1]
```diff
@@ -182,6 +192,9 @@ def __call__(self, batch: SampleBatchType) -> List[SampleBatchType]:
             self.count += batch.agent_steps()
 
         if self.count >= self.min_batch_size:
+            # BONSAI changes begin - added logging
+            logger.info(f"Completed to gather a full batch with size {self.count} with minimum batch size {self.min_batch_size}. Accumulated number of samples is {self.accumulated_count}.")
```
Comment (RuofanKong) [inline rllib/execution/rollout_ops.py:199]: my read is `self.accumulated_count` is always 0?

Comment (Kiko-Aumond) [reply rllib/execution/rollout_ops.py:199]: Fixed.

**Review by RuofanKong — APPROVED** [review]

**Pull request merged**